private/model/api/endpoint_arn.go

```diff
@@ -76,11 +76,15 @@ const accountIDWithARNShapeTmplDef = `
 {{ range $_, $name := $.MemberNames -}}
 	{{ $elem := index $.MemberRefs $name -}}
 	{{ if $elem.AccountIDMemberWithARN -}}
+		{{ $FunctionName := $name }}
+		{{ if eq $name "AccountId" }}
+  		{{ $FunctionName = "AccountID" }}
```

Comment:
we have to change `AccountId` to  `AccountID` to support backwards compatibility. if we wanted to drop it, we could removed this if statement.